### PR TITLE
feat: add new metrics for consistency feature

### PR DESCRIPTION
**Pull request by maudL1n**
1. rename getEtcdNodeKeyDiff to getEtcdConsistencyDiff  
2. refactor  getEtcdConsistencyDiff  waitGroup to errgroup
3. add three consistency metrics , EtcdNodeRevisionDiff  EtcdNodeRaftAppliedIndexDiff   EtcdNodeRaftIndexDiff

### CODE REVIEW

File: pkg/etcd/stats.go [hunk 1]
```diff
@@ -38,8 +39,8 @@ type Stat interface {
 	// Init create etcd client
 	Init(ca, cert, key, endpoint string) error
 
-	// GetTotalKeyNum counts the number of total keys
-	GetTotalKeyNum(keyPrefix string) (uint64, error)
+	// GetTotalConsistencyNum counts the total consistency nums
+	GetTotalConsistencyNum(keyPrefix, endpoint string) (map[string]uint64, error)
```
Comment (tangcong) [inline pkg/etcd/stats.go:43]: keep simple.
GetTotalKeyNum
GetIndex

File: pkg/etcd/stats.go [hunk 2]
```diff
@@ -74,18 +75,45 @@ func (c *StatV3) Init(ca, cert, key, endpoint string) error {
...
-	return uint64(totalKeyNum), nil
+	consistencyMap[featureutil.ConsistencyRevision] = uint64(statusRsp.Header.Revision)
+	consistencyMap[featureutil.ConsistencyRaftRaftAppliedIndex] = statusRsp.RaftAppliedIndex
+	consistencyMap[featureutil.ConsistencyRaftIndex] = statusRsp.RaftIndex
+	return consistencyMap, nil
+}
+
+func (c *StatV2) GetTotalConsistencyNum(keyPrefix, endpoint string) (map[string]uint64, error) {
+	consistencyMap := make(map[string]uint64, 4)
+	klog.V(2).Infof("start to get etcdcluster total consistencyNum")
```
Comment (tangcong) [inline pkg/etcd/stats.go:102]: debug log?

File: pkg/etcd/stats.go [hunk 3]
```diff
@@ -39,7 +40,10 @@ type Stat interface {
 	Init(ca, cert, key, endpoint string) error
 
 	// GetTotalKeyNum counts the number of total keys
-	GetTotalKeyNum(keyPrefix string) (uint64, error)
+	GetTotalKeyNum(ctx context.Context, keyPrefix string) (uint64, error)
+
+	// GetIndex counts the number of total Indexes
```
Comment (tangcong) [inline pkg/etcd/stats.go:45]: // GetIndex gets the etcd metadata index.

File: pkg/inspection/consistency.go [hunk 4]
```diff
@@ -132,26 +133,50 @@ func (c *Server) CollectMemberConsistency(inspection *kstonev1alpha1.EtcdInspect
 	}
 
 	var msg string
-	var nodeKeyDiff uint64
-	ch, errch := c.getEtcdNodeKeyDiff(cluster, path, tlsConfig)
-	err = <-errch
+	nodeConsistencyDiff := make(map[string]uint64)
+	endpintRaftIndexDiff := make(map[string]uint64)
```
Comment (tangcong) [inline pkg/inspection/consistency.go:137]: typo,endpointRaftIndexDiff

File: pkg/inspection/consistency.go [hunk 5]
```diff
@@ -116,42 +118,69 @@ func (c *Server) CollectMemberConsistency(inspection *kstonev1alpha1.EtcdInspect
...
+		msg = fmt.Sprintf("failed to collectEtcdConsistencyDiff, etcd cluster %s, err is %v", cluster.Name, err)
 		klog.Errorf("%s", msg)
 	} else {
-		var keys []uint64
-		for v := range ch {
-			if v == math.MaxUint64 {
-				break
+		endpointRaftIndex := make(map[string]uint64)
+		endpointRaftAppliedIndex := make(map[string]uint64)
+		nodeConsistencyTotal := make(map[string][]uint64)
```
Comment (lianghao208) [inline pkg/inspection/consistency.go:132]:
```go
endpointRaftIndex, endpointRaftAppliedIndex, nodeConsistencyTotal := make(map[string]uint64), make(map[string]uint64), make(map[string]uint64)
```

File: pkg/inspection/consistency.go [hunk 6]
```diff
@@ -116,42 +118,69 @@ func (c *Server) CollectMemberConsistency(inspection *kstonev1alpha1.EtcdInspect
...
+		for endpointConsistency, num := range endpointConsistencyMap {
+			endpointConsistency := strings.Split(endpointConsistency, "_")
+			if len(endpointConsistency) != 2 {
```
Comment (lianghao208) [inline pkg/inspection/consistency.go:135]: If  `endpointConsistency`  can ensure both endpoint and consistencyType are not empty string in method `getEtcdConsistencyDiff()`, this length check won't be necessary.

File: pkg/etcd/stats.go [hunk 7]
```diff
@@ -88,6 +90,20 @@ func (c *StatV3) GetTotalKeyNum(keyPrefix string) (uint64, error) {
 	return uint64(totalKeyNum), nil
 }
 
+func (c *StatV3) GetIndex(ctx context.Context, endpoint string) (map[string]uint64, error) {
+	consistencyMap := make(map[string]uint64, 4)
```
Comment (tangcong) [inline pkg/etcd/stats.go:94]: metadata

Comment (maudL1n) [reply pkg/etcd/stats.go]: change consistencyMap  to metadata

File: pkg/inspection/consistency.go (hunk 6 above)
Comment (maudL1n) [reply pkg/inspection/consistency.go]: deleted this check

File: pkg/inspection/consistency.go (hunk 5 above)
Comment (maudL1n) [reply pkg/inspection/consistency.go]: fixed

**Review by tangcong — APPROVED** [review]

**Review by engow — APPROVED** [review]
LGTM

**Pull request merged**